common/src/buffered-buffer.ts

```diff
@@ -0,0 +1,55 @@
+import { createAsyncQueue } from "./async-queue"
+
+const MAX_BUFFERED_BUFFER = 100;
```

Comment:
It’s possible to use a running total. When you awaiting an enqueue lets you know when it is dequeued. Can maintain a tally.

100 1 byte packets would trigger a disconnect here.